src/Forms/TreeDropdownField.php

```diff
@@ -555,6 +570,27 @@ public function getSourceObject()
         return $this->sourceObject;
     }
 
+    protected function flattenChildrenArray($children, $parentTitles = [])
```

Comment:
Can you document this?

Also, it's not apparent that this method filters by realSearchIds.